SIGSEGV streaming from DJI drone

## Which version are you using?

v0.23.0

## Which operating system are you using?

<!-- fill checkboxes with a x. Example: [x] Linux -->

- [ ] Linux amd64 standard
- [x] Linux amd64 Docker
- [ ] Linux arm64 standard
- [ ] Linux arm64 Docker
- [ ] Linux arm7 standard
- [ ] Linux arm7 Docker
- [ ] Linux arm6 standard
- [ ] Linux arm6 Docker
- [ ] Windows amd64 standard
- [ ] Windows amd64 Docker (WSL backend)
- [ ] macOS amd64 standard
- [ ] macOS amd64 Docker
- [ ] Other (please describe)

## Describe the issue

When publishing an RTMP stream from the DJI Mavic 3 the server crashes. This publishes OK to the nginx mod rtmp.

My speculation is that this because the stream only has video.

## Describe how to replicate the issue

1. start the server
2. publish from the drone
3. Crash

## Did you attach the server logs?

```
2023/05/18 02:51:16 INF MediaMTX / rtsp-simple-server v0.23.0
2023/05/18 02:51:16 INF [RTSP] listener opened on :8554 (TCP)
2023/05/18 02:51:16 INF [RTMP] listener opened on :1935
2023/05/18 02:51:16 INF [HLS] listener opened on :8888
2023/05/18 02:51:16 INF [WebRTC] listener opened on :8889 (HTTP)
2023/05/18 02:51:17 INF [RTMP] [conn 192.168.0.154:49760] opened
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x9d5c80]

goroutine 82 [running]:
github.com/bluenviron/gortsplib/v3/pkg/formats.(*MPEG4AudioGeneric).ClockRate(0xd87ac0?)
	/go/pkg/mod/github.com/bluenviron/gortsplib/v3@v3.6.2/pkg/formats/mpeg4_audio_generic.go:108
github.com/bluenviron/gortsplib/v3.newServerStreamMedia(0xc000546b40, 0xc000534b40, 0x1)
	/go/pkg/mod/github.com/bluenviron/gortsplib/v3@v3.6.2/server_stream_media.go:34 +0x18a
github.com/bluenviron/gortsplib/v3.NewServerStream({0xc0001acd20, 0x2, 0x2})
	/go/pkg/mod/github.com/bluenviron/gortsplib/v3@v3.6.2/server_stream.go:42 +0x129
github.com/bluenviron/mediamtx/internal/core.newStream(0x0?, {0xc0001acd20?, 0x0?, 0x0?}, 0x0?, 0xc0001d9880, {0x7f9ed1090fe8, 0xc0004b0000})
	/s/internal/core/stream.go:30 +0x5e
github.com/bluenviron/mediamtx/internal/core.(*path).sourceSetReady(0xc00062c000, {0xc0001acd20?, 0x0?, 0x0?}, 0x0?)
	/s/internal/core/path.go:641 +0x45
github.com/bluenviron/mediamtx/internal/core.(*path).handlePublisherStart(0xc00062c000, {{0xf6d7a0, 0xc0004b0000}, {0xc0001acd20, 0x2, 0x2}, 0x1, 0xc00008d380})
	/s/internal/core/path.go:795 +0x14b
github.com/bluenviron/mediamtx/internal/core.(*path).run.func2(0xc00062c000)
	/s/internal/core/path.go:473 +0xc98
github.com/bluenviron/mediamtx/internal/core.(*path).run(0xc00062c000)
	/s/internal/core/path.go:499 +0x345
created by github.com/bluenviron/mediamtx/internal/core.newPath
	/s/internal/core/path.go:296 +0x756
```

## Did you attach a network dump?

yes

[dji-rtmp.zip](https://github.com/bluenviron/mediamtx/files/11503944/dji-rtmp.zip)

--- a/README.md
+++ b/README.md
@@ -1217,7 +1217,7 @@ For more advanced options, you can create and serve a custom web page by startin
 
 * WebRTC
   * [WebRTC: Real-Time Communication in Browsers](https://www.w3.org/TR/webrtc/)
-  * [WebRTC Ingestion Protocol (WHIP)](https://datatracker.ietf.org/doc/draft-ietf-wish-whip/)
+  * [WebRTC HTTP Ingestion Protocol (WHIP)](https://datatracker.ietf.org/doc/draft-ietf-wish-whip/)
   * [WebRTC HTTP Egress Protocol (WHEP)](https://datatracker.ietf.org/doc/draft-murillo-whep/)
 
 * Video and audio codecs

--- a/internal/rtmp/tracks/read.go
+++ b/internal/rtmp/tracks/read.go
@@ -44,7 +44,7 @@ func trackFromH264DecoderConfig(data []byte) (formats.Format, error) {
 	}, nil
 }
 
-func trackFromAACDecoderConfig(data []byte) (*formats.MPEG4Audio, error) {
+func trackFromAACDecoderConfig(data []byte) (formats.Format, error) {
 	var mpegConf mpeg4audio.Config
 	err := mpegConf.Unmarshal(data)
 	if err != nil {
@@ -261,10 +261,10 @@ func readTracksFromMetadata(r *message.ReadWriter, payload []interface{}) (forma
 	}
 }
 
-func readTracksFromMessages(r *message.ReadWriter, msg message.Message) (formats.Format, *formats.MPEG4Audio, error) {
+func readTracksFromMessages(r *message.ReadWriter, msg message.Message) (formats.Format, formats.Format, error) {
 	var startTime *time.Duration
 	var videoTrack formats.Format
-	var audioTrack *formats.MPEG4Audio
+	var audioTrack formats.Format
 
 	// analyze 1 second of packets
 outer:

--- a/internal/rtmp/tracks/read_test.go
+++ b/internal/rtmp/tracks/read_test.go
@@ -63,7 +63,7 @@ func TestRead(t *testing.T) {
 			nil,
 		},
 		{
-			"metadata without codec id",
+			"metadata without codec id, video+audio",
 			&formats.H264{
 				PayloadTyp:        96,
 				SPS:               sps,
@@ -81,6 +81,16 @@ func TestRead(t *testing.T) {
 				IndexLength:      3,
 				IndexDeltaLength: 3,
 			},
+		},
+		{
+			"metadata without codec id, video only",
+			&formats.H264{
+				PayloadTyp:        96,
+				SPS:               sps,
+				PPS:               pps,
+				PacketizationMode: 1,
+			},
+			nil,
 		},
 		{
 			"missing metadata, video+audio",
@@ -266,7 +276,7 @@ func TestRead(t *testing.T) {
 				})
 				require.NoError(t, err)
 
-			case "metadata without codec id":
+			case "metadata without codec id, video+audio":
 				err := mrw.Write(&message.DataAMF0{
 					ChunkStreamID:   4,
 					MessageStreamID: 1,
@@ -322,6 +332,57 @@ func TestRead(t *testing.T) {
 					Channels:        flvio.SOUND_STEREO,
 					AACType:         message.AudioAACTypeConfig,
 					Payload:         enc,
+				})
+				require.NoError(t, err)
+
+			case "metadata without codec id, video only":
+				err := mrw.Write(&message.DataAMF0{
+					ChunkStreamID:   4,
+					MessageStreamID: 1,
+					Payload: []interface{}{
+						"@setDataFrame",
+						"onMetaData",
+						flvio.AMFMap{
+							{
+								K: "width",
+								V: float64(2688),
+							},
+							{
+								K: "height",
+								V: float64(1520),
+							},
+							{
+								K: "framerate",
+								V: float64(0o25),
+							},
+						},
+					},
+				})
+				require.NoError(t, err)
+
+				buf, _ := h264conf.Conf{
+					SPS: sps,
+					PPS: pps,
+				}.Marshal()
+
+				err = mrw.Write(&message.Video{
+					ChunkStreamID:   message.VideoChunkStreamID,
+					MessageStreamID: 0x1000000,
+					Codec:           message.CodecH264,
+					IsKeyFrame:      true,
+					Type:            message.VideoTypeConfig,
+					Payload:         buf,
+				})
+				require.NoError(t, err)
+
+				err = mrw.Write(&message.Video{
+					ChunkStreamID:   message.VideoChunkStreamID,
+					MessageStreamID: 0x1000000,
+					Codec:           message.CodecH264,
+					IsKeyFrame:      true,
+					Type:            message.VideoTypeAU,
+					Payload:         []byte{0x01, 0x02, 0x03, 0x04},
+					DTS:             1 * time.Second,
 				})
 				require.NoError(t, err)
 

